stores/Todo.store.ts

```diff
@@ -39,11 +41,17 @@ export class TodoStore {
 
   setTodos = (todos: Todo[]) => {
     this._todos = todos;
+    this.gateway.save(todos);
   };
 
   addTodo = (todo: Todo) => {
     this._todos.push(todo);
+    this.gateway.save(this.todos);
   };
 }
 
 export const TodoContext = createContext<TodoStore | undefined>(undefined);
+
+export function newTodoStore() {
```

Comment:
starting to understand now what is the use of IoC containers like InversifyJS?

Reply:
Sure, they help a tiny bit with maintainability, but I feel like making the client run a library to remove 3 lines of source code is not a valid tradeoff.